artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -345,16 +357,19 @@ public void disconnect(boolean criticalError) {
    }
 
    @Override
-   public boolean checkDataReceived() {
-      boolean res = dataReceived;
-
-      dataReceived = false;
-
-      return res;
+   public void flush() {
+      checkInactivity();
    }
 
-   @Override
-   public void flush() {
+   private void checkInactivity() {
+      if (!this.useKeepAlive) {
```

Comment:
we don't need call super because we don't need to do anything special here, the super will always be called if we don't override it.